### DISCUSSION TITLE
Why is full page writes needed in PolarDB?

### DISCUSSION THREAD

**Discussion by samuel-cc-4451** [polardb/PolarDB-for-PostgreSQL#355] [General]
AFAIK, the idea behind FPWs is to avoid data corruption. Since the Polar filesystem already has a consensus algorithm and stores multiple copies of data, are FPWs actually needed?  

Aurora PostgreSQL doesn't seem to do this: Ref-> https://youtu.be/xrMbzHdPLKM?t=1819

**Comment by fengzunbao** [comment]
PolarDB does not require FPW (Flash Page Writer). PolarDB is based on shared storage, and Read-Write (RW) nodes need to have dirty page control. When an RW node is unable to flush dirty pages in a timely manner, leading to the Buffer Pool being unable to accommodate more pages, it would leverage the FPW mechanism to write pages. However, given the statement that PolarDB doesn't require FPW, it implies that PolarDB utilizes alternative efficient methods within its shared storage architecture to manage buffer pool overflow and ensure data persistence without relying on a traditional FPW process.
